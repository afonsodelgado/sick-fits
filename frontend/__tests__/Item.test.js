import ItemComponent from '../components/Item'
import formatMoney from '../lib/formatMoney'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

const mockItem = {
  id: '123',
  title: 'New Item',
  price: 5000,
  description: 'This is a fake item',
  image: 'item.jpg',
  largerImage: 'largerItem.jpg'
}

describe('<Item />', () => {
  it('renders and displays properly', () => {
    const wrapper = shallow(<ItemComponent item={mockItem}/>)

    expect(wrapper.find('PriceTag').children().text()).toEqual(formatMoney(mockItem.price))
    //For HTML tags we can search straight for the text instead the text on the children
    expect(wrapper.find('Title a').text()).toEqual(mockItem.title)
    expect(wrapper.find('img').props().src).toEqual(mockItem.image)
    expect(wrapper.find('img').props().alt).toEqual(mockItem.title)


  })

  it('renders and match snapshot', () => {
    const wrapper = shallow(<ItemComponent item={mockItem}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});