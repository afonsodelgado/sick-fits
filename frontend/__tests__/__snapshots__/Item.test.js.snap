// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and match snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="New Item"
    src="item.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123",
          },
        }
      }
    >
      <a>
        New Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This is a fake item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="123"
    />
    <DeleteItem
      id="123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
